### PR TITLE
Add Learn Vimscript the Hard Way book

### PR DESCRIPTION
**Pull request by trungqudinh**
## What does this PR do?
Add resource

### Description

Vim is incredibly powerful out of the box, but it doesn't truly shine until you take some time to customize it for your particular work, habits, and fingers. This book will introduce you to Vimscript, the main programming language used to customize Vim. You'll be able to mold Vim into an editor suited to your own personal text editing needs and make the rest of your time in Vim more efficient.

### Why is this valuable (or not)?

This's good for beginer who want to start to go deep in Vim

### How do we know it's really free?

This's free website

### For book lists, is it a book? For course lists, is it a course? etc.

Book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: more/free-programming-cheatsheets.md [hunk 1]
```diff
@@ -82,6 +82,7 @@
 
 * [Editor VI - Guia de Referência](https://aurelio.net/curso/material/vim-ref.html) - Aurelio Marinho Jargas
 * [GNU Emacs Reference Card](https://www.gnu.org/software/emacs/refcards/pdf/refcard.pdf) - GNU.org (PDF)
+* [Learn Vimscript the Hard Way](https://learnvimscriptthehardway.stevelosh.com) - Steve Losh
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:85]: It seems more like a tutorial/book than a cheatsheet due to their size and structure. It have a TOC (read CONTRIBUTING)

Comment (davorpa) [reply more/free-programming-cheatsheets.md:85]: The question now is that is already listed.

https://github.com/EbookFoundation/free-programming-books/blob/521f2e91d2ade5c7df20c361466988136411afb3/books/free-programming-books-langs.md#L2333

Maybe it is a good oportunity to add author and give credits.

**Review by davorpa — CHANGES_REQUESTED** [review]

File: more/free-programming-cheatsheets.md (hunk 1 above)
Comment (trungqudinh) [reply more/free-programming-cheatsheets.md]: Oh, sorry I didn't check it carefully. So I change this to update the author.

**Review by davorpa — APPROVED** [review]
### Ready to merge :fireworks: :tada:

Have a warm welcome :hugs: to `free-programming-books`

**Comment by davorpa** [comment]
Have you a warm welcome :hugs: to `free-programming-books`

**Comment by trungqudinh** [comment]
@davorpa Will you merge this, I dont have permission to do that.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**